### PR TITLE
[Feature] add "renderMonth" support + add Persian locale story

### PR DESCRIPTION
**Pull request by cyclops24**
This PR add `renderMonth` to react-dates and help to add custom calendar like Jalali (Persian) calendar in future.
It's needed for #447 

This is a simple screenshot from story page:
jalali_5

### CODE REVIEW

File: src/components/CalendarMonth.jsx [hunk 1]
```diff
@@ -104,7 +107,7 @@ export default class CalendarMonth extends React.Component {
     } = this.props;
 
     const { weeks } = this.state;
-    const monthTitle = month.format(monthFormat);
+    const monthTitle = renderMonth ? renderMonth(month) : month.format(monthFormat);
```
Comment (ljharb) [inline src/components/CalendarMonth.jsx]: If we're calling a user function here, we'd need to somehow assert that `monthTitle` is a string, to match `month.format()` - and throw if not.

File: stories/SingleDatePicker.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import moment from 'moment';
+import momentJalaali from 'moment-jalaali';
```
Comment (ljharb) [inline stories/SingleDatePicker.js]: Does this need to be added as a dev dep?

Comment (cyclops24) [inline stories/SingleDatePicker.js]: Yes I add it to devDependencies.

File: src/components/CalendarMonth.jsx (hunk 1 above)
Comment (cyclops24) [inline src/components/CalendarMonth.jsx]: So what is your suggestion?
I do this part as a code that now exist in `CalendarDay` component:
https://github.com/airbnb/react-dates/blob/master/src/components/CalendarDay.jsx#L131

Comment (ljharb) [inline src/components/CalendarMonth.jsx]: hm, ok, fair point. this can stay as-is, and we can address them all holistically in the future.

Comment (cyclops24) [inline src/components/CalendarMonth.jsx]: OK, Any other change needs for this PR?

**Comment by cyclops24** [comment]
@majapw do you have any idea about this?

**CC: ** @ljharb

**Pull request merged**